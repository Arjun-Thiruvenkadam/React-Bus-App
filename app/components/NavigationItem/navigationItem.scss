.NavigationItem {
  margin: 10px 0;
  box-sizing: border-box;
  display: block;
  font-family: 'Sacramento', cursive;
  font-size: x-large;
  font-weight: bolder;
  padding: 0px 15px;
}

.NavigationItem a {
  color: #0f3460;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  display: block;
}

.NavigationItem a:hover,
.NavigationItem a:active,
.NavigationItem a.active {
  color: #ddf3f5;
  transform: scale(1.2, 1.2);
}
