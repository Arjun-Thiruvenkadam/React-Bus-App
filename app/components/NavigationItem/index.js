/**
 *
 * NavigationItem
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './navigationItem.scss';

const NavigationItem = props => (
  <li className="NavigationItem">
    <NavLink to={props.link} exact={props.exact} activeClassName="active">
      {props.children}
    </NavLink>
  </li>
);

NavigationItem.propTypes = {
  link: PropTypes.any,
  exact: PropTypes.bool,
  children: PropTypes.any,
};

export default NavigationItem;
