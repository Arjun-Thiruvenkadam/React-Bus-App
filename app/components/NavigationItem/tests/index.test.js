import React from 'react';
import { render } from 'react-testing-library';
import { MemoryRouter } from 'react-router';
import NavigationItem from '../index';

describe('<NavigationItem />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <MemoryRouter>
        <NavigationItem link="/">Link</NavigationItem>
      </MemoryRouter>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
