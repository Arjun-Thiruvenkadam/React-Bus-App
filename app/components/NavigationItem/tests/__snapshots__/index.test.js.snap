// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> should render and match the snapshot 1`] = `
<li
  class="NavigationItem"
>
  <a
    aria-current="page"
    class="active"
    href="/"
  >
    Link
  </a>
</li>
`;
